flink-libraries/flink-table/src/test/scala/org/apache/flink/table/TableEnvironmentTest.scala

```diff
@@ -335,7 +335,8 @@ class TableEnvironmentTest extends TableTestBase {
 
     val table2 = util.addTable[(Long, Int, String)]('d, 'e, 'f)
 
-    val sqlTable2 = util.tEnv.sql(s"SELECT d, e, f FROM $table2 UNION SELECT a, b, c FROM $table")
+    val sqlTable2 = util.tEnv.sql(s"SELECT d, e, f FROM $table2 " +
+        s"UNION ALL SELECT a, b, c FROM $table")
```

Comment:
Yes, you are right. DataStream can only do `UNION ALL`. 
We should prevent the translation of `UNION` and also change the explainString to `union all`, IMO.